### PR TITLE
feat: add tests for hpa analyzer

### PR DESCRIPTION
**Pull request by matthisholleville**
Closes # 

## 📑 Description

New tests have been added to validate the behavior of the hpa analyzer.

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [x] All the tests have passed

### CODE REVIEW

**Review by thschue — APPROVED** [review]
lgtm

File: pkg/analyzer/hpaAnalyzer_test.go [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+	clientset := fake.NewSimpleClientset(
+		&autoscalingv1.HorizontalPodAutoscaler{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:        "example",
+				Namespace:   "default",
+				Annotations: map[string]string{},
+			},
+			Spec: autoscalingv1.HorizontalPodAutoscalerSpec{
+				ScaleTargetRef: autoscalingv1.CrossVersionObjectReference{
+					Kind: "unsuported",
```
Comment (AlexsJones) [inline pkg/analyzer/hpaAnalyzer_test.go:83]: Spelling

File: pkg/analyzer/hpaAnalyzer_test.go [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+		&autoscalingv1.HorizontalPodAutoscaler{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:        "example",
+				Namespace:   "default",
+				Annotations: map[string]string{},
+			},
+			Spec: autoscalingv1.HorizontalPodAutoscalerSpec{
+				ScaleTargetRef: autoscalingv1.CrossVersionObjectReference{
+					Kind: "Deployment",
+					Name: "unexistent",
```
Comment (AlexsJones) [inline pkg/analyzer/hpaAnalyzer_test.go:130]: non-existent

**Review by AlexsJones — APPROVED** [review]
Some spelling fixes if you have time

**Comment by matthisholleville** [comment]
Sure. I Will change.

**Review by thschue — APPROVED** [review]
lgtm

**Pull request merged**